pkg/generate/app/componentmatch.go

```diff
@@ -0,0 +1,86 @@
+package app
+
+import (
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+)
+
+// ComponentMatch is a match to a provided component
+type ComponentMatch struct {
+	Value       string
+	Argument    string
+	Name        string
+	Description string
+	Score       float32
+	Insecure    bool
+
+	Builder     bool
+	Image       *imageapi.DockerImage
+	ImageStream *imageapi.ImageStream
+	ImageTag    string
+	Template    *templateapi.Template
+}
+
+func (m *ComponentMatch) String() string {
+	return m.Argument
+}
+
+// IsImage returns whether or not the component match is an
+// image or image stream
+func (m *ComponentMatch) IsImage() bool {
+	return m.Template == nil
+}
+
+// IsTemplate returns whether or not the component match is
+// a template
+func (m *ComponentMatch) IsTemplate() bool {
+	return m.Template != nil
+}
+
+// Checks if the ComponentMatch is an exact match
```

Comment:
All comments of exported functions in this file need to start with the function names